Framework/Geometry/src/Instrument/DetectorInfo.cpp

```diff
@@ -228,6 +228,52 @@ DetectorInfo::signedTwoTheta(const std::pair<size_t, size_t> &index) const {
   return angle;
 }
 
+double DetectorInfo::azimuthal(const size_t index) const {
+  if (isMonitor(index))
+    throw std::logic_error("Azimuthal angle is not defined for monitors");
+
+  const auto samplePos = samplePosition();
+  const auto beamLine = samplePos - sourcePosition();
+
+  if (beamLine.nullVector()) {
+    throw Kernel::Exception::InstrumentDefinitionError(
+        "Source and sample are at same position!");
+  }
+
+  const auto sampleDetVec = position(index) - samplePos;
+  const auto horizontal =
+      m_instrument->getReferenceFrame()->vecPointingHorizontal();
+
+  const double dotHorizontal = sampleDetVec.scalar_prod(horizontal);
+  const auto vertical = Kernel::normalize(beamLine).cross_prod(horizontal);
+  const double dotVertical = sampleDetVec.scalar_prod(vertical);
+
+  return atan2(dotVertical, dotHorizontal);
+}
+
+double DetectorInfo::azimuthal(const std::pair<size_t, size_t> &index) const {
+  if (isMonitor(index))
+    throw std::logic_error("Azimuthal angle is not defined for monitors");
+
+  const auto samplePos = samplePosition();
+  const auto beamLine = samplePos - sourcePosition();
+
+  if (beamLine.nullVector()) {
+    throw Kernel::Exception::InstrumentDefinitionError(
+        "Source and sample are at same position!");
+  }
+
+  const auto sampleDetVec = position(index) - samplePos;
+  const auto horizontal =
+      m_instrument->getReferenceFrame()->vecPointingHorizontal();
+
+  const double dotHorizontal = sampleDetVec.scalar_prod(horizontal);
+  const auto vertical = Kernel::normalize(beamLine).cross_prod(horizontal);
+  const double dotVertical = sampleDetVec.scalar_prod(vertical);
+
+  return atan2(dotVertical, dotHorizontal);
+}
```

Comment:
This is the same code as for the other `azimuthal`, can you move it into a helper method templated on `index`?

Reply:
Many of the accessors in `DetectorInfo` would benefit from templating. I was trying to keep this one small.

Reply:
You can make it even smaller using a template?

Reply:
All of the functions would benefit from the template. I will do that in a separate PR.

Reply:
It turns out that the use of initializer lists makes type deduction in the templated version too hard for the compiler to deduce.